### PR TITLE
[Refactor] refactor SortDesc in sort

### PR DESCRIPTION
**Pull request by murphyatwork**
## What type of PR is this：
- [ ] bugfix
- [ ] feature
- [ ] enhancement
- [x] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #

## Problem Summary(Required) ：

Merge the `sort_order` and `null_first` into a `SortDesc` to make it more simple.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] I have added user document for my new feature or new function

### CODE REVIEW

**Comment by mofeiatwork** [comment]
run starrocks_admit_test

**Comment by mofeiatwork** [comment]
run starrocks_admit_test

**Review by kangkaisen — APPROVED** [review]

File: be/src/exec/vectorized/sorting/sorting.h [hunk 1]
```diff
@@ -66,7 +66,16 @@ struct SortDesc {
     int null_first;
 
     SortDesc() = default;
+    SortDesc(bool is_asc, bool inull_first) {
+        sort_order = is_asc ? 1 : -1;
+        null_first = (inull_first ? -1 : 1) * sort_order;
```
Comment (ZiheLiu) [inline be/src/exec/vectorized/sorting/sorting.h:71]: How about using initialization list to init them? Otherwise, they will be assigned twice.

**Review by ZiheLiu — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

File: be/src/exec/vectorized/sorting/sorting.h (hunk 1 above)
Comment (mofeiatwork) [reply be/src/exec/vectorized/sorting/sorting.h:71]: Almost no cost... I choose to make it more clear instead of performance.

**Pull request merged**